assets/prototype/domain/eventEditor/datetimes/dateList/filterBar/useDatesListFilterState.test.ts

```diff
@@ -0,0 +1,324 @@
+		reserved: 0,
+		sold: 76,
+		startDate: '2009-12-18T11:31:00+00:00',
+		status: 'ACTIVE',
+		__typename: 'EspressoDatetime',
+	},
+];
+
+test('useDatesListFilterState result', () => {
+	const { result } = renderHook(() => useDatesListFilterState(datetimes));
+
+	expect(Object.keys(result.current).length).toBe(8);
+
+	expect(result.current.dates).toEqual(expect.arrayContaining(datetimes));
+	expect(typeof result.current.datesSortedBy).toBe('string');
+	expect(typeof result.current.displayDates).toBe('string');
+
+	expect(typeof result.current.setDatesSortedBy).toBe('function');
+	expect(typeof result.current.setDisplayDates).toBe('function');
+	expect(typeof result.current.setShowDates).toBe('function');
+});
+
+test('should update datesSortedBy by invoking setDatesSortedBy with corresponding accepted enums', () => {
+	const { result } = renderHook(() => useDatesListFilterState(datetimes));
+
+	act(() => {
+		result.current.setDatesSortedBy(DatesSortedBy.byName);
+	});
+	expect(result.current.datesSortedBy).toBe('byName');
+
+	act(() => {
+		result.current.setDatesSortedBy(DatesSortedBy.byId);
+	});
+	expect(result.current.datesSortedBy).toBe('byId');
+
+	act(() => {
+		result.current.setDatesSortedBy(DatesSortedBy.byOrder);
+	});
+	expect(result.current.datesSortedBy).toBe('byOrder');
+
+	act(() => {
+		result.current.setDatesSortedBy(DatesSortedBy.chronologically);
+	});
+	expect(result.current.datesSortedBy).toBe('chronologically');
+});
+
+test('should update displayDates by invoking setDisplayDates with corresponding accepted enums', () => {
+	const { result } = renderHook(() => useDatesListFilterState(datetimes));
+
+	act(() => {
+		result.current.setDisplayDates(DisplayDates.start);
+	});
+	expect(result.current.displayDates).toBe('start');
+
+	act(() => {
+		result.current.setDisplayDates(DisplayDates.end);
+	});
+	expect(result.current.displayDates).toBe('end');
+
+	act(() => {
+		result.current.setDisplayDates(DisplayDates.both);
+	});
+	expect(result.current.displayDates).toBe('both');
+});
+
+test('should update showDates by invoking setShowDates with corresponding accepted enums', () => {
+	const { result } = renderHook(() => useDatesListFilterState(datetimes));
+
+	act(() => {
+		result.current.setShowDates(ShowDates.all);
+	});
+	expect(result.current.showDates).toBe('all');
+
+	act(() => {
+		result.current.setShowDates(ShowDates.upcomingOnly);
+	});
+	expect(result.current.showDates).toBe('upcomingOnly');
+});
+
+test('should update processedDates to reflect changes made by invoking setDatesSortedBy', () => {
+	const { result } = renderHook(() => useDatesListFilterState(datetimes));
+
+	act(() => {
+		result.current.setDatesSortedBy(DatesSortedBy.chronologically);
+	});
+	const startDates = R.map(R.view(R.lensProp('startDate')))(result.current.processedDates);
+	const expectedstartDates = [
+		'2000-01-12T08:00:00+00:00',
+		'2009-12-18T11:31:00+00:00',
+		'2019-12-18T11:31:00+00:00',
+		'2020-01-12T08:00:00+00:00',
+	];
```

Comment:
Instead of writing that manually, why not get that data from `datetimes` array and sort it before comparing it with the resultant data?